test/common/stream_info/filter_state_impl_test.cc

```diff
@@ -77,6 +77,25 @@ TEST_F(FilterStateImplTest, Simple) {
   EXPECT_EQ(1u, destruction_count);
 }
 
+TEST_F(FilterStateImplTest, SharedPointerAccessor) {
+  size_t access_count = 0u;
+  size_t destruction_count = 0u;
+  filter_state().setData(
+      "test_name", std::make_shared<TestStoredTypeTracking>(5, &access_count, &destruction_count),
+      FilterState::StateType::Mutable, FilterState::LifeSpan::FilterChain);
+  EXPECT_EQ(0u, access_count);
+  EXPECT_EQ(0u, destruction_count);
+
+  auto obj = filter_state().getDataSharedMutableGeneric("test_name");
+  EXPECT_EQ(5, dynamic_cast<TestStoredTypeTracking*>(obj.get())->access());
+  EXPECT_EQ(1u, access_count);
+  EXPECT_EQ(0u, destruction_count);
+
+  resetFilterState();
+  EXPECT_EQ(1u, access_count);
+  EXPECT_EQ(0u, destruction_count);
```

Comment:
when will the destruction count be incremented for shared pointer version?  why doesn't resetFilterState() increment it?

Reply:
When the shared pointer is deleted on block exit. This allows sharing objects between two streams without one stream deleting it.

Reply:
makes sense, mind adding a reset to obj shared_ptr, then checking the `destruction_count` is incremented as expect? Thanks

Reply:
Done.